### PR TITLE
fix segmentation fault caused by container removal.

### PR DESCRIPTION
**Pull request by enginmanap**
This fixes issue #2

### CODE REVIEW

**Comment by feimosi** [comment]
Great, thanks!
I haven't used C++ for quite some time, could you explain briefly what was the issue?

**Comment by enginmanap** [comment]
The issue is zmienna variable is destructed after return call. Since item is part of it, it is destructed also. Graph class tries to access it to create a copy but it is not there anymore so causing segmentation fault. There is a compile warning about it too.

The issue is, my commit does not actually fix it. Compiler realize zmienna is not used, as an optimization removes that variable and uses heap[0].item instead. Since heap is not local it is not destroyed. As you can guess, this is not a correct usage, and it would fail on different compilers or environments. I have a better one. That one returns an object instead of pointer. The object is copied over at returns so it should actually be safe.

File: Yet-Another-Pathfinding-Simulator/PriorityQueue.h [hunk 1]
```diff
@@ -72,14 +72,16 @@ namespace yaps
...
+        T removeHighestPrior()
         {
-            if (NextFree == 0) return NULL;
-            Node zmienna = heap[0];
+            if (NextFree == 0) throw std::exception();
+            T returnVal = heap[0].item;
+
             heap[0] = heap[--NextFree];
-            heap.erase(heap.begin() + NextFree);
+            heap.erase(heap.begin());
```
Comment (feimosi) [inline Yet-Another-Pathfinding-Simulator/PriorityQueue.h]: Why `+ NextFree` is removed?

**Review by feimosi — COMMENTED** [review]
Alright, understood. Just one last question below.

File: Yet-Another-Pathfinding-Simulator/PriorityQueue.h (hunk 1 above)
Comment (enginmanap) [reply Yet-Another-Pathfinding-Simulator/PriorityQueue.h]: thats my mistake. Let me fix that up too.

**Comment by enginmanap** [comment]
fix that too

**Comment by feimosi** [comment]
Ok, now looks good, thanks for the contribution!

**Comment by enginmanap** [comment]
thank you, glad to help.

**Pull request merged**